### DISCUSSION TITLE
Kafka + Apicurio testing

### DISCUSSION THREAD

**Discussion by CROprogrammer** [Apicurio/apicurio-registry#3428] [Q&A]
Hi,
what tools are good for performance and functional testing of Kafka processor that uses Apicurio schema registry?

**Comment by EricWittmann** [comment]
I don't know that there are any specific tools designed to test Kafka client applications (which is what I assume you're asking about).  If you want to know about testing Java applications in general (a Kafka client application would be a Java application typically) then JRebel has a decent blog post (if a bit dated now):

https://www.jrebel.com/blog/best-java-performance-testing-tools-and-technologies

**Reply by CROprogrammer** [reply]
I'm looking for a tool that would help me performance test producing messages (serialization - interaction with schema registry) and consuming messages (deserialization - again interaction with schema registry).
My goal is to test how much impact on Kafka client applications, using schema registry has
